### PR TITLE
feat: add minimum commission rate to x/staking

### PR DESCRIPTION
**Pull request by youngjoon-lee**
### Important
This PR is based on the `v0.42-panacea` branch which points to the v0.42.11 tag. Although the current panacea-core v2.0.2 uses cosmos-sdk v0.42.9, I thought it's okay to use cosmos-sdk v0.42.11 because there's no serious breaking changes. Also, if we decided to implement this feature based on v0.42.9, the branch strategy may become complicated.

### Changes
- Added a new `min_commission_rate` genesis param of the `x/staking` module.
- Modified the logic `CreateValidator` and `UpdateValidator` to check if the desired commission rate is `>= min_commission_rate`.
- Added a special function for updating validator commissions without checking `max_commission_change_rate`.
   - Our chain upgrade strategy is updating the commission rate of all validators whose commission rate is smaller than `min_commission_rate`. In normal mode, this logic can cause an error: `commission change rate cannot be more than the max rate`. That's why we need a special function which doesn't ignore the `max_commission_change_rate` param.

### Tips for reviewers
You don't need to review changes which are displayed as `Some generated files are not rendered` because those file are generated by `make proto-gen`. Unfortunately, Cosmos guys didn't execute `make proto-gen` after modifying proto files. That's why so many unexpected `.pb.go` files have been regenerated in this PR.

### References
- https://github.com/osmosis-labs/cosmos-sdk/commit/c3f499c413d51f1b1a534b61d46de68e19db326f
- https://github.com/osmosis-labs/cosmos-sdk/commit/7dd164f811893e76c49aae1dd6497884afc388db
- https://github.com/osmosis-labs/cosmos-sdk/commit/07832f29b69513b28a55b453b6a904c3bd128710
- My change is slightly different from Osmosis' change (especially `MustUpdateValidatorCommission()`). I've considered more corner cases when upgrading the chain with updating commission rates.

### What's next?
Please jump to https://github.com/medibloc/panacea-core/pull/291.

### CODE REVIEW

File: Makefile
Comment (youngjoon-lee) [inline Makefile:14]: I specified the version explicitly because the `buf > 0.56.0` has so many CLI breaking changes. It's safe to use 0.56.0 since this Makefile was written when the latest version of buf was 0.56.0.

File: x/staking/keeper/validator.go
Comment (gyuguen) [inline x/staking/keeper/validator.go:164]:
```suggestion
		return commission, fmt.Errorf("cannot set validator commission to larger than 100%")
```

Comment (youngjoon-lee) [reply x/staking/keeper/validator.go:164]: I had to escape the `%` with `%` because the `fmt.Errorf()` understands `%` as a special character for string interpolation. 
https://go.dev/play/p/zys4wDhEe3p
If it looks weird, I can replace `fmt.Errorf()` with `errors.New()` but I didn't because of the consistency with the next line.

**Review by gyuguen — APPROVED** [review]
LGTM 👍

**Comment by gyuguen** [comment]
Ah, I have one question.
It seems that this min commission is not supported by cosmos-sdk.
Should this change be applied later when upgrading the cosmos-sdk version?

**Comment by youngjoon-lee** [comment]
@gyuguen 
> It seems that this min commission is not supported by cosmos-sdk.

Yeah. That's why I forked `cosmos/cosmos-sdk` as `medibloc/cosmos-sdk`.

> Should this change be applied later when upgrading the cosmos-sdk version?

After merging this PR, we will release the `v0.42.11-panacea` tag from this `medibloc/cosmos-sdk` repository. Then, we will change the cosmos-sdk dependency of `panacea-core` from `cosmos/cosmos-sdk v0.42.9` to `medibloc/cosmos-sdk v0.42.11-panacea`. That have been already done in the PR https://github.com/medibloc/panacea-core/pull/291.
If you're worried about the data market protocol that is being developed in the `master` branch of `panacea-core`, it would be all good if we use `medibloc/cosmos-sdk v0.42.11-panacea` in the `master` branch as well. 
In fact, we have to use the `medibloc/cosmos-sdk` instead of `cosmos/cosmos-sdk` forever as Osmosis does.

**Comment by youngjoon-lee** [comment]
I merged this PR even though I got only one approval, in order to submit the software upgrade gov proposal soon.
But, please revisit this PR if you have any question or suggestion.

**Comment by youngjoon-lee** [comment]
As Osmosis does in their [osmosis-labs/cosmos-sdk](https://github.com/osmosis-labs/cosmos-sdk) forked repo, it would be okay to not update the `CHANGELOG.md` because of potential merge-conflicts with the upstream repo. Instead, I will write the release note in detail at GitHub Release pages.

**Pull request merged**
